src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -152,19 +136,25 @@ public void setEnableSchematization(final boolean enableSchematization) {
    * @return a Row wrapper which contains both actual content(payload) and metadata
    */
   private SnowflakeTableRow processRecord(SinkRecord record) {
+    SnowflakeRecordContent valueContent;
+
     if (record.value() == null || record.valueSchema() == null) {
-      throw SnowflakeErrors.ERROR_5016.getException();
-    }
-    if (!record.valueSchema().name().equals(SnowflakeJsonSchema.NAME)) {
-      throw SnowflakeErrors.ERROR_0009.getException();
-    }
-    if (!(record.value() instanceof SnowflakeRecordContent)) {
-      throw SnowflakeErrors.ERROR_0010.getException(
-          "Input record should be SnowflakeRecordContent object");
+      if (this.ingestTombstoneRecords) {
```

Comment:
This is the crux of the PR - if we want to ingestTombstoneRecords, create an empty record